packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -2601,6 +2606,19 @@ export class ContainerRuntime
 
 		this.verifyNotClosed();
 
+		// Reference Sequence Number may be about to change, and it must be consistent across a batch, so flush now
+		this.outbox.flush();
+
+		this.ensureNoDataModelChanges(() => {
+			this.processImpl(messageCopy, local);
+		});
+	}
+
+	/**
+	 * Implementation of core logic for {@link ContainerRuntime.process}, once preconditions are established
+	 * @param messageCopy - Shallow copy of the sequenced message
+	 */
+	private processImpl(messageCopy: ISequencedDocumentMessage, local: boolean): void {
```

Comment:
nit: Not sure what this naming is supposed to mean. Is it "process implementation"?